stereo/src/stereo_adaptive_cost_so.cpp

```diff
@@ -36,167 +36,188 @@
-      {
-        fwd[x][d] = acc[x][d] - c_min + std::min (std::min (fwd[x-1][d], fwd[x-1][d-1] + static_cast<float> (smoothness_weak_)), std::min (fwd[x-1][d+1] + static_cast<float> (smoothness_weak_), c_min + static_cast<float> (smoothness_strong_)));
-      } 
-      fwd[x][max_disp_-1] = acc[x][max_disp_-1] - c_min + std::min (fwd[x-1][max_disp_-1], std::min(fwd[x-1][max_disp_-2] + static_cast<float> (smoothness_weak_), c_min + static_cast<float> (smoothness_strong_)));
-    }//x
- 
-    //Backward
+        if (fwd[x - 1][d] < c_min)
+          c_min = fwd[x - 1][d];
+
+      fwd[x][0] =
+          acc[x][0] - c_min +
+          std::min(fwd[x - 1][0],
+                   std::min(fwd[x - 1][1] + static_cast<float>(smoothness_weak_),
+                            c_min + static_cast<float>(smoothness_strong_)));
+      for (int d = 1; d < max_disp_ - 1; d++) {
+        fwd[x][d] =
+            acc[x][d] - c_min +
+            std::min(std::min(fwd[x - 1][d],
+                              fwd[x - 1][d - 1] + static_cast<float>(smoothness_weak_)),
+                     std::min(fwd[x - 1][d + 1] + static_cast<float>(smoothness_weak_),
+                              c_min + static_cast<float>(smoothness_strong_)));
+      }
+      fwd[x][max_disp_ - 1] =
+          acc[x][max_disp_ - 1] - c_min +
+          std::min(
+              fwd[x - 1][max_disp_ - 1],
+              std::min(fwd[x - 1][max_disp_ - 2] + static_cast<float>(smoothness_weak_),
+                       c_min + static_cast<float>(smoothness_strong_)));
+    } // x
+
+    // Backward
     for (int d = 0; d < max_disp_; d++)
-      bck[width_-1][d] = acc[width_-1][d];
- 
-    for (int x = width_-2; x > max_disp_ + x_off_; x--)
-    {
-      
-      float c_min = bck[x+1][0];
+      bck[width_ - 1][d] = acc[width_ - 1][d];
+
+    for (int x = width_ - 2; x > max_disp_ + x_off_; x--) {
+
+      float c_min = bck[x + 1][0];
       for (int d = 1; d < max_disp_; d++)
-       if (bck[x+1][d] < c_min)
-        c_min = bck[x+1][d];
- 
-      bck[x][0] =  acc[x][0] - c_min + std::min (bck[x+1][0], std::min (bck[x+1][1] + static_cast<float> (smoothness_weak_), c_min + static_cast<float> (smoothness_strong_)));
-       for (int d = 1; d < max_disp_ - 1; d++)
-       bck[x][d] = acc[x][d] - c_min + std::min (std::min(bck[x+1][d], bck[x+1][d-1] + static_cast<float> (smoothness_weak_)), std::min (bck[x+1][d+1] + static_cast<float> (smoothness_weak_), c_min + static_cast<float> (smoothness_strong_)));
-      bck[x][max_disp_-1] = acc[x][max_disp_-1] - c_min + std::min (bck[x+1][max_disp_-1], std::min (bck[x+1][max_disp_-2] + static_cast<float> (smoothness_weak_), c_min + static_cast<float> (smoothness_strong_)));
-    }//x
- 
-    //last scan
-    for (int x = x_off_ + max_disp_ + 1; x < width_; x++)
-    {
-      float c_min = std::numeric_limits<float>::max ();
+        if (bck[x + 1][d] < c_min)
+          c_min = bck[x + 1][d];
+
+      bck[x][0] =
```

Comment:
Might need to refactor this to be more understandable later.

Reply:
Same here :)